src/foam/nanos/cron/SimpleIntervalSchedule.js

```diff
@@ -56,104 +57,76 @@
         class: 'foam.u2.view.DayOfWeekView',
       },
       visibility: function(frequency) {
-        if ( frequency != 'Week' )
-           return foam.u2.DisplayMode.HIDDEN;
+        if ( frequency != this.TimeUnit.WEEK )
+          return foam.u2.DisplayMode.HIDDEN;
         return foam.u2.DisplayMode.RW;
       }
     },
     {
+      class: 'Enum',
+      of: 'foam.nanos.cron.MonthlyChoice',
       name: 'monthlyChoice',
       label: '',
-      view: {
-       class: 'foam.u2.view.RadioView',
-       choices: [
-         [1, 'On The...'],
-         [2, 'Each']
-        ],
-       isHorizontal: true
+      view: function(_, X) {
+        return {
+          isHorizontal: true,
+          class: 'foam.u2.view.RadioEnumView'
+        };
       },
       visibility: function(frequency) {
-        if ( frequency != 'Month')
+        if ( frequency != this.TimeUnit.MONTH )
           return foam.u2.DisplayMode.HIDDEN;
         return foam.u2.DisplayMode.RW;
       }
     },
     {
       name: 'dayOfMonth',
       label: '',
-      //will be replaced by multiselect dayofmonth view
       view: {
-        class: 'foam.u2.view.ChoiceView',
-        choices: [
-          1,
-          2
-        ]
+        class: 'foam.u2.view.DayOfMonthView'
       },
       visibility: function(monthlyChoice, frequency) {
-        if ( frequency != 'Month' )
+        if ( frequency != this.TimeUnit.MONTH )
+          return foam.u2.DisplayMode.HIDDEN;
+        if ( monthlyChoice != this.MonthlyChoice.EACH )
           return foam.u2.DisplayMode.HIDDEN;
-        if ( monthlyChoice != 2 )
-           return foam.u2.DisplayMode.HIDDEN;
         return foam.u2.DisplayMode.RW;
       }
     },
     {
-    //first , second , third ...
-      name: 'vagueFreq',
+      // first , second , third ...
+      class: 'Enum',
+      of: 'foam.nanos.cron.SymbolicFrequency',
+      name: 'symbolicFrequency',
       label: '',
-      view: {
-        class: 'foam.u2.view.ChoiceView',
-        choices: [
-          'First',
-          'Second',
-          'Third',
-          'Before Last',
-          'Last'
-        ]
-      },
       gridColumns: 6,
       visibility: function(monthlyChoice, frequency ) {
-        if ( frequency != 'Month' )
+        if ( frequency != this.TimeUnit.MONTH )
+          return foam.u2.DisplayMode.HIDDEN;
+        if ( monthlyChoice !=  this.MonthlyChoice.ON_THE )
           return foam.u2.DisplayMode.HIDDEN;
-        if ( monthlyChoice !=  1 )
-           return foam.u2.DisplayMode.HIDDEN;
         return foam.u2.DisplayMode.RW;
       }
     },
     {
+      class: 'Enum',
+      of: 'foam.time.DayOfWeek',
```

Comment:
Missing support for "Weekday" in DayOfWeek, can we extend Enums? Otherwise it will have to be a new Enum I assume

Reply:
What does weekday indicate - any day or all days?  How does weekday work?

Reply:
That depends on how we treat days that are not possible either because of the rails not being available on those days or if the dates don't exist in that month. Since symbolic frequencies can only be used with months I don't think we would run into the issue of last day always being Sunday. Our solution originally was to just add a disclaimer message explaining how we treat this exception cases and keep their handling consistent across the board. If we still follow that I think we can stick with last day. @tharmaman did I miss something?